### DISCUSSION TITLE
Should we release 0.69.1?

### DISCUSSION THREAD

**Discussion by cortinico** [reactwg/react-native-releases#23] [Patches]
## Should we release 0.69.1?

Current Release: 0.69.0

Conversations on this thread are limited:

- [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking).
- [qualified cherry-pick requests](https://reactnative.dev/contributing/release-faq#what-is-a-qualified-pick-request) of commits on main that [did not make the previous patch version](https://reactnative.dev/contributing/release-faq#how-do-i-know-if-my-fixfeature-is-in-a-certain-release).

Please include a link to the specific commit on main to be cherry-picked, for example: [facebook/react-native@bd2b7d6](https://github.com/facebook/react-native/commit/20b0eba581a00e5e7e300f6377379b836617c147)

In other words, if you cannot point to a particular commit on main, then your request likely belongs as a new issue.
If the issue is a [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking), please reference the issue here.

---

#### List of qualified picks

- https://github.com/facebook/react-native/commit/114d31feeeb47f5a57419e5088c3cbe9340f757a
- https://github.com/facebook/react-native/commit/43f831b23caf22e59af5c6d3fdd62fed3d20d4ec
- https://github.com/facebook/react-native/commit/c2088e1267439fd60d9e6abc8992d9ca3c2d7018
- https://github.com/facebook/react-native/commit/f97c6a5b498eec95e99a02c7842cb2ae160cd6cd
- https://github.com/facebook/react-native/commit/79baca678a743560fa16fdd551f1d0d018d34304
- https://github.com/facebook/react-native/commit/4ea38e16bf533955557057656cba5346d2372acd

To be rolled-over to 0.69.2:
- https://github.com/facebook/react-native/commit/0bd523955385a3b1e622077b7ee4ea0df3c5c158
- https://github.com/facebook/react-native/commit/9f112af50abbcd059e3872ae63143c35bb8c0f7f
- https://github.com/facebook/react-native/pull/34026

#### Local commits to backport to main

1.

**Comment by AlexanderEggers** [comment]
After updating to 0.69.0 I have multiple packages failing with the message which causes my build to fail:

```
warn Package @sentry/react-native has been ignored because it contains invalid configuration. Reason: "dependency.platforms.ios.sharedLibraries" is not allowed
warn Package expo has been ignored because it contains invalid configuration. Reason: "dependency.platforms.ios.podspecPath" is not allowed
warn Package react-native-vector-icons has been ignored because it contains invalid configuration. Reason: "dependency.assets" is not allowed
warn Package react-native-flipper has been ignored because it contains invalid configuration. Reason: "dependency.platforms.ios.project" is not allowed
```

When downgrading to 0.68.x, the warnings are gone and I can compile correctly again.

**Reply by swrobel** [reply]
Seeing the same thing here. This is causing my build to fail as well:
```
warn Package @bugsnag/react-native has been ignored because it contains invalid configuration. Reason: "dependency.platforms.ios.sharedLibraries" is not allowed
warn Package @fortawesome/free-brands-svg-icons has been ignored because it contains invalid configuration. Reason: Cannot find module 'node_modules/@fortawesome/free-brands-svg-icons/package.json.js'
warn Package @fortawesome/free-solid-svg-icons has been ignored because it contains invalid configuration. Reason: Cannot find module 'node_modules/@fortawesome/free-solid-svg-icons/package.json.js'
warn Package @fortawesome/pro-duotone-svg-icons has been ignored because it contains invalid configuration. Reason: Cannot find module 'node_modules/@fortawesome/pro-duotone-svg-icons/package.json.js'
warn Package @fortawesome/pro-light-svg-icons has been ignored because it contains invalid configuration. Reason: Cannot find module 'node_modules/@fortawesome/pro-light-svg-icons/package.json.js'
warn Package @fortawesome/pro-regular-svg-icons has been ignored because it contains invalid configuration. Reason: Cannot find module 'node_modules/@fortawesome/pro-regular-svg-icons/package.json.js'
warn Package @fortawesome/pro-solid-svg-icons has been ignored because it contains invalid configuration. Reason: Cannot find module 'node_modules/@fortawesome/pro-solid-svg-icons/package.json.js'
warn Package react-native-blob-util has been ignored because it contains invalid configuration. Reason: "dependency.hooks" is not allowed
warn Package react-native-flipper has been ignored because it contains invalid configuration. Reason: "dependency.platforms.ios.project" is not allowed
```

**Reply by kelset** [reply]
wait, they are warnings but they are making a build fail? Something is not right here :(

**Reply by tgensol** [reply]
Same here 

```
warn Package @bugsnag/react-native has been ignored because it contains invalid configuration. Reason: "dependency.platforms.ios.sharedLibraries" is not allowed
warn Package instabug-reactnative has been ignored because it contains invalid configuration. Reason: "dependency.hooks" is not allowed
warn Package react-native-adjust has been ignored because it contains invalid configuration. Reason: "dependency.platforms.ios.podspecPath" is not allowed
warn Package react-native-vector-icons has been ignored because it contains invalid configuration. Reason: "dependency.assets" is not allowed
warn Package react-native-flipper has been ignored because it contains invalid configuration. Reason: "dependency.platforms.ios.project" is not allowed
```

**Reply by tido64** [reply]
Yeah, these props got removed in CLI 8.x as part of the `link` command being deprecated and other refactorings. There was a discussion about it here: https://github.com/reactwg/react-native-releases/discussions/21#discussioncomment-2985774

For most props, you can simply remove them and you should be fine. For the `assets` prop, see the discussion I linked to.

**Reply by thymikee** [reply]
As @tido64 mentioned. I'll look into ignoring those warnings (which are in fact errors from schema validation which prevent building the `config` object) from the CLI side

**Reply by kelset** [reply]
@thymikee so it sounds like a new version of CLI v.8.x.y will be needed to address this?

**Reply by thymikee** [reply]
@kelset yup. I'm trying to prioritize this, but can't just jump on it asap

**Reply by kelset** [reply]
oh yeah no rush - I'm just trying to understand where and how the fix will need to happen 🤗

**Reply by RnbWd** [reply]
it's not including these libraries, even though it's a warning

**Comment by SymntxHomendra51** [comment]
fix the problems with the flipper.

**Reply by fortmarek** [reply]
Hey!

Could you specify your exact issue? Is it [this](https://github.com/facebook/react-native/issues/33764) or something different?

**Reply by SymntxHomendra51** [reply]
Hello Mate!
I'm referring to this problem. Dont know which issue does it belongs though
https://rnfirebase.io/#altering-cocoapods-to-use-frameworks
![image](https://user-images.githubusercontent.com/95675342/175337835-617f9c7e-ec6f-4385-bec9-cac1f24dc2a2.png)

**Reply by lukebars** [reply]
@SymntxHomendra51 this is open source software, so usually nobody is being paid to do the work. You can always help by funding somebody to do the work or help yourself. Otherwise - just be patient, sorry. 🙏

**Reply by mikehardy** [reply]
Yep - I literally wrote that note @SymntxHomendra51 - and there are patches already for 0.69.0 and Hermes (they are in the pick list above). 

But for Flipper I'm not sure it is being tracked, and it does not have a PR yet - you could be the savior and post the PR! Same for New Architecture - details here - https://github.com/reactwg/react-native-releases/discussions/21#discussioncomment-2924919

Please note that **for this discussion** we need commit hashes to pick. That's it. So for purposes of this discussion there is no action here yet unless/until you or someone posts a fix to Flipper that is merged to main and you comment here with the commit hash to pick. That would be fantastic - then we'll be in business

**Reply by mikehardy** [reply]
To be more prescriptive perhaps:

1) need an issue logged with Flipper such that Flipper works with `use_frameworks! :linkage => :dynamic` and `use_frameworks! :linkage => :static` - and then the associated PR produced by someone affected + motivated
2) need an issue logged with react-native to get `use_frameworks! :linkage => :dynamic` (and `:static`) working with New Architecture enabled, note this likely needs to be driven internally by Meta as it's a cross-cutting change, see https://github.com/invertase/react-native-firebase/issues/6323#issuecomment-1164660774 specifically
3) wait for 0.69.1 such that existing commits already in pick list are released, so you may use react-native-firebase v15+ along with react-native 0.69.1 (with or without hermes, but still no flipper or new architecture until 1 and 2 here are done)

@SymntxHomendra51 those are actions you can take - if you would do those and cross-link them here I will personally track them across releases etc and help where possible.

**Reply by mikehardy** [reply]
react-native-firebase v15 works (with hermes!) with react-native 0.69.0 if you apply [this patch-package patch](https://github.com/mikehardy/rnfbdemo/blob/main/patches/react-native%2B0.69.0.patch) - which is itself the product of @kudo - but now it is easy to digest and I've qualified the two pick requests, it looks like it's good. That was item 3 in my previous comment. Items 1 and 2 are still work to do

**Comment by AlirezaHadjar** [comment]
Hi,
according to the changelog, the link command and react-native-config.js were removed. So how should we add a font to our project? should we do it manually or there is another way?

**Reply by tido64** [reply]
Please see this discussion: https://github.com/reactwg/react-native-releases/discussions/21#discussioncomment-2985774

**Comment by opayen** [comment]
We would like to have https://github.com/facebook/react-native/commit/114d31feeeb47f5a57419e5088c3cbe9340f757a cherry-picked to 0.69.1.
It is a fix to a regression that was introduced in 0.68 (see https://github.com/facebook/react-native/issues/33977)

**Reply by fortmarek** [reply]
Yes, it's already in the list of commits to pick, thanks 🙌

**Comment by kelset** [comment]
let's not forget https://github.com/facebook/react-native/pull/34026 cc @mikehardy @Arkkeeper

**Reply by cortinico** [reply]
ETA for merging this is Monday/Tuesday next week. I'll add it to the list of pick

**Comment by oleksandr-dziuban** [comment]
Hi guys, please review a critical issues for Android 12 please
Native crash for Animated Nodes
https://github.com/facebook/react-native/issues/33375

**Reply by oleksandr-dziuban** [reply]
It affects all RN applications on Android 12. A lot of apps are crashing.... we don't have a workaround yet..

**Reply by mikehardy** [reply]
> we don't have a workaround yet

It's worse than that, there is not even a reproduction. First step: reproduction. Second step: PR against react-native@main merged. Third Step: join discussion here on whether it is appropriate to cherry-pick the commit from second step

Discussing it here is unfortunately quite premature. I posted on your issue though just to offer that there may be hope for a repro if you try samsung's device test lab. Get a commit hash from a merged commit related to that issue then this is the spot

**Comment by leotm** [comment]
just following up to not forget
- https://github.com/reactwg/react-native-releases/discussions/21#discussioncomment-2941715

https://github.com/facebook/react-native/commit/0bd523955385a3b1e622077b7ee4ea0df3c5c158 looks like the worthy one
(https://github.com/facebook/react-native/commit/9f112af50abbcd059e3872ae63143c35bb8c0f7f a follow-up commit)

worth noting the minor [template update](https://github.com/facebook/react-native/commit/0bd523955385a3b1e622077b7ee4ea0df3c5c158#diff-fcc02228077c73cb58800911f68c7de9938b9848279ee6caa73c27dd63d27212) there ahead of time too
which will land on upgrade-helper on ship

**Reply by cortinico** [reply]
This had to be backported in a different PR. Kudos to @cipolleschi for doing so here:
- https://github.com/facebook/react-native/pull/34098

Will land in 0.69.2

**Comment by kelset** [comment]
surfacing here, seems there's an ongoing issue with Yoga: https://github.com/facebook/react-native/issues/33976 (cc @Bardiamist)

**Reply by kelset** [reply]
did a PR about that: https://github.com/facebook/react-native/pull/34058

**Reply by kelset** [reply]
there's a better PR for it out now: https://github.com/facebook/react-native/pull/34064 let's focus on trying to get this second one in and then decide what to do about mine

**Reply by kelset** [reply]
commit landed: https://github.com/facebook/react-native/commit/4ea38e16bf533955557057656cba5346d2372acd

**Reply by fortmarek** [reply]
Thanks, already being released now 🚀

**Comment by mikehardy** [comment]
Partial pick of https://github.com/facebook/react-native/commit/118cf689142d6754a300d714caf33b4fd64b575d
Specifically the part that mirrors React pick here https://github.com/facebook/react/commit/6bf3deef5
With goal of ingesting a "react-shallow-renderer" package [that supports React 18](https://github.com/enzymejs/react-shallow-renderer/commits/master), since react-native is on React 18 now

This will quiet an install warning (for example when you run `yarn`)

> warning "react-native > react-shallow-renderer@16.14.1" has incorrect peer dependency "react@^16.0.0 || ^17.0.0".

I think the only way to actually do this is to just [bump that dependency in package.json](https://github.com/facebook/react-native/blob/0.69-stable/package.json#L123) to 16.15.0 (and regenerate lock files if lock files are committed? [Looks like they are](https://github.com/facebook/react-native/blob/0.69-stable/yarn.lock)) on the 0.69 branch, but the commit does exist on main already at least

**Reply by cortinico** [reply]
I'm a bit hesitant on the partial pick. If the impact is just the warning at `yarn install` time, I would vote against this.

**Reply by kelset** [reply]
I agree, I'd rather do a local commit to just do the bump of that dependecy if it's a big problem. But it's just a warning right?

**Reply by tido64** [reply]
The warning can actually cause install failures with certain setups of npm or pnpm. I'd rather fix it if it's not too much of a hassle.

**Reply by Titozzz** [reply]
FYI, tested on npm 7.20.3 (no particular setup)
```
~/Documents/GitHub/RN069
❯ npm i
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: react-shallow-renderer@16.14.1
npm WARN Found: react@18.0.0
npm WARN node_modules/react
npm WARN   react@"18.0.0" from the root project
npm WARN   3 more (react-native, react-test-renderer, use-sync-external-store)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react@"^16.0.0 || ^17.0.0" from react-shallow-renderer@16.14.1
npm WARN node_modules/react-shallow-renderer
npm WARN   react-shallow-renderer@"16.14.1" from react-native@0.69.0
npm WARN   node_modules/react-native
npm WARN   1 more (react-test-renderer)
npm WARN 
npm WARN Conflicting peer dependency: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@"^16.0.0 || ^17.0.0" from react-shallow-renderer@16.14.1
npm WARN   node_modules/react-shallow-renderer
npm WARN     react-shallow-renderer@"16.14.1" from react-native@0.69.0
npm WARN     node_modules/react-native
npm WARN     1 more (react-test-renderer)
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added
npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated uglify-es@3.3.9: support for ECMAScript is superseded by `uglify-js` as of v3.13.0

added 1281 packages, and audited 1282 packages in 39s

96 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```

-> It works even if there is a huge warning.

**Reply by tido64** [reply]
Reading the docs, it sounds like it would only fail if the project also has a direct dependency on `react-shallow-renderer`. Can you try with `--strict-peer-deps`?

**Reply by mikehardy** [reply]
Sure seems like just bumping that one dep would be an easy+friendly thing to do :thinking:

**Reply by Titozzz** [reply]
@tido64 

FYI --strict-peer-deps fails for other reasons 😅 

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: @react-native-community/eslint-config@2.0.0
npm ERR! Found: eslint@7.32.0
npm ERR! node_modules/eslint
npm ERR!   dev eslint@"^7.32.0" from the root project
npm ERR!   peer eslint@">=6" from @react-native-community/eslint-config@2.0.0
npm ERR!   node_modules/@react-native-community/eslint-config
npm ERR!     dev @react-native-community/eslint-config@"^2.0.0" from the root project
npm ERR!   2 more (@typescript-eslint/eslint-plugin, @typescript-eslint/parser)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! babel-eslint@"^10.1.0" from @react-native-community/eslint-config@2.0.0
npm ERR! node_modules/@react-native-community/eslint-config
npm ERR!   dev @react-native-community/eslint-config@"^2.0.0" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: eslint@8.18.0
npm ERR! node_modules/eslint
npm ERR!   peer eslint@">= 4.12.1" from babel-eslint@10.1.0
npm ERR!   node_modules/babel-eslint
npm ERR!     babel-eslint@"^10.1.0" from @react-native-community/eslint-config@2.0.0
npm ERR!     node_modules/@react-native-community/eslint-config
npm ERR!       dev @react-native-community/eslint-config@"^2.0.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --no-strict-peer-deps, --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/thibault/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/thibault/.npm/_logs/2022-06-29T15_48_34_217Z-debug.log
```

**Reply by cortinico** [reply]
> Sure seems like just bumping that one dep would be an easy+friendly thing to do 🤔

Can someone do this and send a PR against the release branch? We could potentially include this in 0.69.2

**Reply by mikehardy** [reply]
I'll see if I can do a PR against the release branch really quickly. This just caused a total failure for me in the `npx react-native init` case :weary: 

```

mike@kunashir:~/work/react-random % npx react-native init RealDeviceTest
Need to install the following packages:
  react-native@0.69.1
Ok to proceed? (y) 
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: undefined@undefined
npm ERR! Found: react-native@0.68.2
npm ERR! node_modules/react-native
npm ERR!   peer react-native@"*" from @react-native-community/cli@7.0.3
npm ERR!   node_modules/@react-native-community/cli
npm ERR!     @react-native-community/cli@"^7.0.3" from react-native@0.68.2
npm ERR!   react-native@"0.69.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! react-native@"0.69.1" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: react@18.0.0
npm ERR! node_modules/react
npm ERR!   peer react@"18.0.0" from react-native@0.69.1
npm ERR!   node_modules/react-native
npm ERR!     react-native@"0.69.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/mike/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/mike/.npm/_logs/2022-06-29T20_28_45_100Z-debug-0.log
mike@kunashir:~/work/react-random % npx --version
8.12.0
mike@kunashir:~/work/react-random % node --version
v16.15.1
```

(doing `npx --legacy-peer-deps react-native init NoOneIsReadingThisCommentAreThey?` worked but, that hurts a bit)

**Reply by mikehardy** [reply]
@cortinico  :eyes: --> https://github.com/facebook/react-native/pull/34104 :smile:
